### PR TITLE
Data scalling encoding

### PR DESCRIPTION
**Pull request by Conrad-Push**
I have added 2 functions for scalling and encoding data columns. In fact, currently only encoding function has been added to the main.py script. The scalling function may be used in the future with comparing the different approaches to the binary classificator models training :)

### CODE REVIEW

File: data_preprocessing/main.py [hunk 1]
```diff
@@ -4,6 +4,7 @@
 def main():
     columns_to_filter = ["TotPkts", "TotBytes", "SrcBytes", "flag", "service", "count", "dst_bytes", "class"]
     columns_to_transform_to_log = ["TotPkts", "SrcBytes", "dst_bytes"]
+    columns_to_encode = ["Flag", "Service", "Class"]
```
Comment (pploszczyca) [inline data_preprocessing/main.py]: The encode part is before changing names to PascalCase, so when this part is executed it has old column names. It has to be switched to old names or moved encoding part after `data_processor.change_column_names_to_pascal_case()` execution

File: data_preprocessing/data_preprocessor.py [hunk 2]
```diff
@@ -28,6 +28,14 @@ def choose_columns(self, columns: list[str]):
     def transform_column_data_to_logarithmic_scale(self, column: str):
         pass
 
+    @abstractmethod
+    def scaling_column_data_numerical_attributes(self, column: str):
+        pass
```
Comment (pploszczyca) [inline data_preprocessing/data_preprocessor.py:33]: <s>Is this function used somewhere?</s> I read a PR description and I see that this function is added for the future :sweat_smile: 
Do you tested if it works properly?

File: data_preprocessing/pandas_data_preprocessor.py [hunk 3]
```diff
@@ -40,6 +41,14 @@ def save_data(self, folder_path):
     def transform_column_data_to_logarithmic_scale(self, column: str):
         self.data[column] = np.log(self.data[column] + 1)
 
+    def scaling_column_data_numerical_attributes(self, column: str):
+        scaler = StandardScaler()
+        self.data[column] = scaler.fit_transform(self.data[column])
+
+    def encoding_column_data_categorical_attributes(self, column: str):
+        label_encoder = LabelEncoder()
+        self.data[column] = label_encoder.fit_transform(self.data[column])
```
Comment (pploszczyca) [inline data_preprocessing/pandas_data_preprocessor.py]: [Sugestion] Probably `StandardScaler` and `LabelEncoder` could be initialize once in __init__ method

```python
    def __init__(self, data_type='general'):
        super().__init__()
        logging.basicConfig(level=logging.INFO,
                            format='%(asctime)s - %(name)s - %(levelname)s - %(data_type)s - %(message)s')
        self.logger = logging.getLogger(__name__)
        self.data_type = data_type
        self.scaler = StandardScaler()
        self.label_encoder = LabelEncoder()
```

File: requirements.txt [hunk 4]
```diff
@@ -1,4 +1,5 @@
 pandas>=2.1.2
 numpy>=1.26.1
-seaborn>=0.13.1
+seaborn>=0.13.0
 matplotlib>=3.8.1
+scikit-learn
```
Comment (pploszczyca) [inline requirements.txt]: [Sugestion] It could be good to add `scikit-learn` minimum version, like we have for other dependencies

**Review by pploszczyca — COMMENTED** [review]
If it comes to problem with running this script, the problem can occur when you try to run it from another folder.
```
> python3 data_preprocessing/main.py 
Traceback (most recent call last):
  File "/home/piotrek/Projekty/Cyber-Semestr-2/PracowniaProblemowa/TrafficGuard/data_preprocessing/main.py", line 1, in <module>
    from data_preprocessing.pandas_data_preprocessor import PandasDataPreprocessor
ModuleNotFoundError: No module named 'data_preprocessing'
```

If you use the command line, you have to enter this folder directly and execute this code. 
```
> cd data_preprocessing
> python3 main.py 
```

Also I suppose `data_preprocessing/__init__.py` isn't needed here

File: data_preprocessing/data_preprocessor.py (hunk 2 above)
Comment (Conrad-Push) [reply data_preprocessing/data_preprocessor.py:33]: Yes, unfortunately due to this problem with 'No module named...' while trying to run main.py, I had to test it on Google Colab :P

File: data_preprocessing/main.py (hunk 1 above)
Comment (Conrad-Push) [reply data_preprocessing/main.py]: I've changed the column names to the old ones

File: data_preprocessing/pandas_data_preprocessor.py (hunk 3 above)
Comment (Conrad-Push) [reply data_preprocessing/pandas_data_preprocessor.py]: Good point, our code will be better organized

File: requirements.txt (hunk 4 above)
Comment (Conrad-Push) [reply requirements.txt]: Done

**Comment by Conrad-Push** [comment]
I don't know what is going on with this "No module named 'data_preprocessing'" but it still does not work even running it with your suggestion @pploszczyca  ://

**Review by pploszczyca — APPROVED** [review]
Good job :beers:

**Review by CodeByJacob — APPROVED** [review]

**Review by biongoo — APPROVED** [review]

**Pull request merged**